clustering/common/src/main/java/org/jboss/as/clustering/controller/AbstractCapabilityReference.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.clustering.controller;
+
+import org.jboss.as.controller.CapabilityReferenceRecorder;
+import org.jboss.as.controller.OperationContext;
+import org.wildfly.clustering.service.Requirement;
+
+/**
+ * Abstract {@link CapabilityReferenceRecorder} containing logic common to attribute and resource capability references
+ * @author Paul Ferraro
+ */
+public abstract class AbstractCapabilityReference implements CapabilityReferenceRecorder {
+
+    private final Capability capability;
+    private final Requirement requirement;
+
+    protected AbstractCapabilityReference(Capability capability, Requirement requirement) {
+        this.capability = capability;
+        this.requirement = requirement;
+    }
+
+    @Override
+    public String getBaseDependentName() {
+        return this.capability.getName();
+    }
+
+    @Override
+    public String getBaseRequirementName() {
+        return this.requirement.getName();
+    }
+
+    protected String getDependentName(OperationContext context) {
```

Comment:
Passing the resource and address as params instead of OperationContext removes the need to manage capabilities in a specific way from the RemoveHandler

Reply:
Is this comment meant to be specific to the getDependentName(...) method?  This method doesn't need the Resource to generate its name, only the path address.